pandas/tseries/offsets.py

```diff
@@ -1122,21 +1097,21 @@ def rule_code(self):
 
     @apply_wraps
     def apply(self, other):
-        n = self.n
```

Comment:
After #18877 and #18875, the randomized testing I'm running locally is turning up bugs (pytz.NonExistentTimeError notwithstanding) exclusively in SemiMonthOffset and FY5253Quarter.